src/serialization/interval_log/mod.rs

```diff
@@ -0,0 +1,588 @@
+        IntervalLogHeaderWriter {
+            internal_writer: InternalLogWriter {
+                writer,
+                serializer,
+                text_buf: String::new(),
+                serialize_buf: Vec::new(),
+            },
+        }
+    }
+
+    /// Add a comment line.
+    ///
+    /// If you do silly things like write a comment with a newline character, you'll end up with
+    /// an un-parseable log file. Don't do that.
+    pub fn write_comment(&mut self, s: &str) -> io::Result<()> {
+        self.internal_writer.write_comment(s)
+    }
+
+    /// Write a StartTime log line. See the module-level documentation for more info.
+    ///
+    /// This should only be called once to avoid creating a confusing interval log.
+    pub fn write_start_time(&mut self, seconds_since_epoch: f64) -> io::Result<()> {
+        self.internal_writer.write_fmt(format_args!(
+            "#[StartTime: {:.3} (seconds since epoch)]\n",
+            seconds_since_epoch
+        ))
+    }
+
+    /// Write a BaseTime log line. See the module-level documentation for more info.
+    ///
+    /// This should only be called once to avoid creating a confusing interval log.
+    pub fn write_base_time(&mut self, seconds_since_epoch: f64) -> io::Result<()> {
+        self.internal_writer.write_fmt(format_args!(
+            "#[BaseTime: {:.3} (seconds since epoch)]\n",
+            seconds_since_epoch
+        ))
+    }
+
+    /// Once you're finished with headers, convert this into a log writer so you can write interval
+    /// histograms.
+    pub fn into_log_writer(self) -> IntervalLogWriter<'a, 'b, W, S> {
+        IntervalLogWriter {
+            internal_writer: self.internal_writer,
+        }
+    }
+}
+
+/// Writes interval histograms in an interval log.
+///
+/// This isn't created directly; start with an `IntervalLogHeaderWriter`. Once you've written the
+/// headers and ended up with an `IntervalLogWriter`, typical usage would be to write a histogram
+/// at regular intervals (e.g. once a second).
+pub struct IntervalLogWriter<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> {
+    internal_writer: InternalLogWriter<'a, 'b, W, S>,
+}
+
+impl<'a, 'b, W: 'a + io::Write, S: 'b + Serializer> IntervalLogWriter<'a, 'b, W, S> {
+    /// Add a comment line.
+    pub fn write_comment(&mut self, s: &str) -> io::Result<()> {
+        self.internal_writer.write_comment(s)
+    }
+
+    /// Write an interval histogram.
+    ///
+    /// `start_timestamp` is the time since the epoch in seconds. If you're using a StartTime or
+    /// BaseTime offset, you should instead use a delta since that time. See the discussion about
+    /// timestamps in the module-level documentation.
+    ///
+    /// `duration` is the duration of the interval in seconds.
+    ///
+    /// `tag` is an optional tag for this histogram.
+    ///
+    /// `max_value_divisor` is used to scale down the max value to something that may be more human
```

Comment:
Maybe this should be fixed for the entire interval log? It seems like it can't be a good idea to allow different divisors for different intervals.

Reply:
Yeah, I think that sounds like a good idea. *And* it fits nicely with the proposed builder pattern!